main/readme.md

```diff
@@ -0,0 +1,19 @@
+This is a kitchen-sink Nix flake for any packages that we want to define
+for use at Freckle.
+
+## Package versioning
+
+Since this flake is intended for use across many components and teams, package
+names should include version numbers to prepare for the possibility of the flake
+supporting multiple versions of a package simultaneously.
+
+Packages should generally be aliased in several ways so that each use site
+can choose what level of specificity in the fourmolu version it wants.
+For example, we have:
+
+- `fourmolu.default` -
+  If you don't care what version of fourmolu you get
+- `fourmolu.version-0-13-x` -
+  If you want major version 0.13 but don't care what minor version
+- `fourmolu.version-0-13-0` -
```

Comment:
Not that I really care, but more just curious: is there a limitation that prevents these from being named `fourmolu`, `fourmolu.0.13` and `fourmolu.0.13.0`?

Reply:
The nix language does let you put `.` (or anything at all) in attribute keys as long at you write them in quotes, but I'm not sure to what extent the command-line interface can deal with it. I think at the very least dealing with them would be inconvenient.

Reply:
I'm not sure what conventions apply. Nixpkgs generally doesn't keep multiple versions of a package around and so it doesn't do this at all. When they do, it's pretty inconsistent. The way they do GHC is horrifying e.g. expressing 9.2.7 as "ghc927". My habit is to err on the side of not abbreviating things, but happy to collapse "version" to "v" as that's pretty standard.